modules/arrow/src/schema/geoarrow-utils.ts

```diff
@@ -0,0 +1,522 @@
+  const schema = arrowTable.schema;
+  for (let i = 0; i < schema.fields.length; i++) {
+    const field = schema.fields[i];
+    const metadata = field.metadata;
+    if (metadata && metadata.has(GEOARROW_COLUMN_METADATA_KEY)) {
+      const geoEncoding = metadata.get(GEOARROW_COLUMN_METADATA_KEY);
+      geometryColumns.set(field.name, {encoding: geoEncoding});
+    }
+  }
+  return geometryColumns;
+}
+
+/**
+ * get geoarrow encoding from geoarrow column
+ */
+export function getGeoArrowEncoding(
+  schema: Schema,
+  geometryColumnName: string 
+): string {
+  const field = schema.fields.find(field => field.name === geometryColumnName);
+  return field?.metadata?.get(GEOARROW_COLUMN_METADATA_KEY) || '';
+}
+
+/**
+ *  update bounds from geoarrow samples
+ *
+ * @param flatCoords the flattend coordinates array from one chunk of geoarrow column
+ * @param nDim the number of dimensions of the coordinates
+ * @param bounds the bounds to be updated
+ * @param sampleSize how many samples to be used to update the bounds, default is 1000 per chunk
+ */
+function updateBoundsFromGeoArrowSamples(
+  flatCoords: Float64Array,
+  nDim: number,
+  bounds: [number, number, number, number],
+  sampleSize: number = 100
+): void {
+  const numberOfFeatures = flatCoords.length / nDim;
+  const sampleStep = Math.max(Math.floor(numberOfFeatures / sampleSize), 1);
+
+  for (let i = 0; i < numberOfFeatures; i += sampleStep) {
+    const lng = flatCoords[i * nDim];
+    const lat = flatCoords[i * nDim + 1];
+    if (lng < bounds[0]) {
+      bounds[0] = lng;
+    }
+    if (lat < bounds[1]) {
+      bounds[1] = lat;
+    }
+    if (lng > bounds[2]) {
+      bounds[2] = lng;
+    }
+    if (lat > bounds[3]) {
+      bounds[3] = lat;
+    }
+  }
+}
+
+// binary geometry template, see deck.gl BinaryGeometry
+const BINARY_GEOMETRY_TEMPLATE = {
+  globalFeatureIds: {value: new Uint32Array(0), size: 1},
+  positions: {value: new Float32Array(0), size: 2},
+  properties: [],
+  numericProps: {},
+  featureIds: {value: new Uint32Array(0), size: 1}
+};
+
+type BinaryGeometryContent = {
+  featureIds: Uint32Array;
+  flatCoordinateArray: Float64Array;
+  nDim: number;
+  geomOffset: Int32Array;
+  geometryIndicies: Uint16Array;
+};
+
+/**
+ * get binary polygons from geoarrow polygon column
+ * @param chunk one chunk of geoarrow polygon column
+ * @param geoEncoding the geo encoding of the geoarrow polygon column
+ * @returns BinaryGeometryContent
+ */
+function getBinaryPolygonsFromChunk(chunk: Data, geoEncoding: string): BinaryGeometryContent {
+  const isMultiPolygon = geoEncoding === GEOARROW_ENCODINGS.MULTI_POLYGON;
+
+  const polygonData = isMultiPolygon ? chunk.children[0] : chunk;
+  const ringData = polygonData.children[0];
+  const pointData = ringData.children[0];
+  const coordData = pointData.children[0];
+  const nDim = pointData.stride;
+  const geomOffset = ringData.valueOffsets;
+  const flatCoordinateArray = coordData.values;
```

Comment:
For arbitrary geoarrow input, you should assert that the low-level coords are interleaved, not separated (i.e. a struct of x and y columns) because geoarrow allows both in its spec

Reply:
Good suggestion! Will update. Thanks!